base/number.jl

```diff
@@ -168,10 +168,23 @@ abs(x::Real) = ifelse(signbit(x), -x, x)
 
 Squared absolute value of `x`.
 
+For complex numbers this is faster than `abs(x)^2`.
+
+See also [`abs`](@ref), [`conj`](@ref), [`real`](@ref).
+
 # Examples
 ```jldoctest
 julia> abs2(-3)
 9
+
+julia> abs2(3.0 + 4.0im)
+25.0
+
+julia> sum(abs2, [1+2im, 3+4im])
+30
+
+julia> abs2.(Int8.(8:12)) |> Tuple  # overflow, since 12 * 12 > typemax(Int8)
```

Comment:
If this has to be shown, imo, just show the `abs2` applied to the `12`. Introducing more advanced concepts like broadcasting and `|>` syntax for the examples of very simple functions is likely to be more confusing than helping.

Also, does the overflow case has to be explicitly shown at all? It isn't even shown in e.g. `+`.

Reply:
I'd argue that it should be shown for `+` since we get at least 1 question a month on it.

Reply:
I'm not sure it need be shown here, this was mostly a response to people saying "`abs` shows this"  but it's non-obvious there.

Reply:
Now removed